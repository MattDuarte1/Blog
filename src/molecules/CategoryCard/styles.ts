import styled, { css, DefaultTheme } from 'styled-components';

const MediaContainer = {
  small: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      width: 80%;
    }
  `,
};

interface ContainerProps {
  isActive: boolean;
}

const MenuIsActive = (isActive: boolean, theme: DefaultTheme) =>
  isActive &&
  css`
    background-color: ${theme.colors.blue};
    color: #fff;
  `;

export const Container = styled.article<ContainerProps>`
  ${({ theme, isActive }) => css`
    display: flex;
    align-items: center;
    padding-left: ${theme.spacings.large};
    width: 220px;
    height: 280px;
    color: ${theme.colors.black};
    background-color: ${theme.colors.white};
    filter: drop-shadow(4px 6px 13px rgba(0, 0, 0, 0.1));
    border-radius: 8px;
    cursor: pointer;

    > div {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      text-align: center;
      padding: 0px;
      gap: 30px;
    }

    &:hover {
      background-color: ${theme.colors.blue};
      color: white;
    }

    ${MediaContainer.small(theme)}
    ${MenuIsActive(isActive, theme)}
  `}
`;
