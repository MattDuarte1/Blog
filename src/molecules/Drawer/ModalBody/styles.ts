import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    flex: 1 1 0%;
    display: flex;
    padding: 2rem;
    width: 100%;
    height: inherit;
    justify-content: flex-start;
    gap: 2rem;
    flex-direction: column;
    align-items: center;

    > a {
      min-width: 100%;

      &:hover {
        color: ${theme.colors.blue};
      }
    }
  `}
`;
