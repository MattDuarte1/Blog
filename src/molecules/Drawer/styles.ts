import styled, { css } from 'styled-components';

interface IModalProps {
  isOpen?: boolean;
}

export const ModalBackDrop = styled.div<IModalProps>`
  ${({ isOpen }) => css`
    position: fixed;
    right: 0px;
    top: 0px;
    width: 100vw;
    transform: translateX(${isOpen ? '0%' : '100%'});
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.48);
    z-index: 100;
    transition: all 0.8s ease-in-out;
  `}
`;

export const Modal = styled.div<IModalProps>`
  ${({ theme, isOpen }) => css`
    position: fixed;
    right: 1px;
    top: 0px;
    bottom: 0px;
    width: ${isOpen ? '285px' : '0px'};
    padding: 25px;
    height: inherit;
    background-color: ${theme.colors.white};
    z-index: 110;
    transition: all 1s ease;

    > svg {
      position: absolute;
      top: 10px;
      right: 2rem;

      font-size: ${theme.font.sizes.large};
      color: ${theme.colors.black};
      cursor: pointer;
      transition: all 0.5s ease;

      &:hover {
        color: ${theme.colors.blue};
        transform: scale(1.1);
      }
    }
  `}
`;
