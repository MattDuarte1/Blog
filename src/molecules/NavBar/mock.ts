export default [
  { title: 'Inicio', newtab: true, href: '#header' },
  { title: 'Categoria', newtab: false, href: '#categoria' },
  { title: 'Sobre', newtab: false, href: '#sobre' },
  { title: 'Procurar', newtab: false, href: '#procurar' },
];
