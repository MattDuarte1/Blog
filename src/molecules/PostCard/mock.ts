import { IPost } from '@/interfaces/Post';

const Data = {
  id: '95380751',
  title: 'Vue JS: O que Ã©, como funciona ?',
  image: {
    url: 'https://www.datocms-assets.com/90479/1674243496-vuejs.svg',
    alt: 'vuejs svg',
  },
  createdAt: '2023-02-03T15:48:45+00:00',
  authorCompact: {
    name: 'Mateus Duarte',
    image: {
      url: 'https://www.datocms-assets.com/90479/1673505399-67812250_126640048653747_8066872119595701241_n.jpg',
      alt: 'author image',
    },
  },
} as IPost;

export default Data;
