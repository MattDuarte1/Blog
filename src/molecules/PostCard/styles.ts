import styled, { css, DefaultTheme } from 'styled-components';

const MediaBreapoints = {
  AuthorDescription: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        > img {
          width: 47px;
          height: 47px;
        }
      }
    `,
  },
};

export const Container = styled.article`
  ${({ theme }) => css`
    width: 300px;
    height: 425px;
    padding: 1.4rem;
    background: ${theme.colors.white};
    position: relative;

    box-shadow: 4px 6px 13px rgba(0, 0, 0, 0.1);
    border-radius: 12px;
    text-align: start;
    transition: all 0.2s ease;
    cursor: pointer;

    > h1 {
      margin-left: 1rem;
    }

    img {
      margin-bottom: 2rem;
    }

    &:hover {
      transform: scale(1.02);
      filter: drop-shadow(0 1.5px 2px rgba(0, 0, 0, 0.2));
    }
  `}
`;

export const AuthorContent = styled.div`
  ${({ theme }) => css`
    position: absolute;
    padding: 0 1.4rem;
    width: 100%;
    height: 57px;
    left: 0px;
    top: 348px;
    display: flex;
    color: ${theme.colors.black};
  `}
`;

export const AuthorDescription = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 90%;
    flex-direction: row;
    justify-content: center;
    padding: 0px;
    gap: 15px;
    color: ${theme.colors.black};

    ${MediaBreapoints.AuthorDescription['small'](theme)}
  `}
`;

export const Avatar = styled.div`
  overflow: hidden;
  border-radius: 50%;
  width: 50px;
  height: 50px;
`;

export const AuthorInfo = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 2px;
  gap: 4.2px;

  width: 184px;
  height: 46.2px;
`;

export const ViewBox = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
  flex-flow: column nowrap;

  > svg {
    color: ${({ theme }) => theme.colors.black};
    font-size: ${({ theme }) => theme.font.sizes.medium};
  }

  > h1 {
    display: inline;
  }
`;
