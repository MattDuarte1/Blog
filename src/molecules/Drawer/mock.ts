import { theme } from '@/styles/theme';

export default {
  state: {
    drawerIsOpen: false,
    theme: theme,
  },
  menus: [
    { title: 'Inicio', newtab: false, href: '/#header' },
    { title: 'Sobre', newtab: false, href: '/#hero' },
    { title: 'Categoria', newtab: false, href: '/#categoria' },
    { title: 'Posts', newtab: false, href: '/#posts' },
  ],
} as const;
