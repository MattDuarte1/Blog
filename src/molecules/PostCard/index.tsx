import Heading from '@/atoms/Heading';
import * as Styled from './styles';
import Image from 'next/image';
import { AiOutlineEye } from 'react-icons/ai';
import { IPost } from '@/interfaces/Post';
import { forwardRef } from 'react';

type PostCardProps = IPost & {
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
};

const PostCard = forwardRef<HTMLElement, PostCardProps>(
  ({ title, authorCompact, image, createdAt, id, onClick, ...props }, ref) => {
    const newFormatDataBR = new Date(createdAt).toLocaleDateString();

    return (
      <Styled.Container onClick={onClick} id={id} ref={ref} {...props}>
        <Image src={image.url} alt={image.alt} width={260} height={198} />
        <Heading size="xsmall" color="black">
          {title}
        </Heading>
        <Styled.AuthorContent>
          <Styled.AuthorDescription>
            <Styled.Avatar>
              <Image
                width={50}
                height={50}
                priority
                src={authorCompact.image.url}
                alt={authorCompact.image.alt}
              />
            </Styled.Avatar>
            <Styled.AuthorInfo>
              <h4>{authorCompact.name}</h4>
              <p>{newFormatDataBR}</p>
            </Styled.AuthorInfo>
          </Styled.AuthorDescription>
          <Styled.ViewBox>
            <AiOutlineEye />
            <Heading size="small">0</Heading>
          </Styled.ViewBox>
        </Styled.AuthorContent>
      </Styled.Container>
    );
  },
);

PostCard.displayName = 'PostCard Component';

export default PostCard;
