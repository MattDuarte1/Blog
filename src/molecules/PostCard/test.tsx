import { screen } from '@testing-library/react';
import PostCard from '.';
import { renderTheme } from '@/styles/render-theme';
import { theme } from '@/styles/theme';
import mock from '@/organisms/PostSection/mock';

describe('<PostCard>', () => {
  it('Should render a PostCard component', () => {
    const { container } = renderTheme(
      <PostCard {...mock} onClick={jest.fn()} />,
    );

    const postCard = container.firstElementChild;

    expect(container.firstElementChild).toBeInTheDocument();
    expect(postCard).toHaveStyle({
      width: '300px',
      height: '425px',
      padding: '1.4rem',
      background: theme.colors.white,
      position: 'relative',
      'box-shadow': '4px 6px 13px rgba(0,0,0,0.1)',
      'border-radius': '12px',
      'text-align': 'start',
      cursor: 'pointer',
    });
    expect(screen.getByRole('heading', { name: mock.title }));
    expect(postCard.querySelector('h4').innerHTML).toBe(
      mock.authorCompact.name,
    );
  });

  it('Should match snapshot component', () => {
    renderTheme(<PostCard {...mock} onClick={jest.fn()} />);

    expect(screen.getByRole('article')).toMatchSnapshot();
  });
});
