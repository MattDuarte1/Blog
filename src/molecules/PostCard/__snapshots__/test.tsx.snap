// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard> Should match snapshot component 1`] = `
.c1 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 1.617rem;
  text-transform: none;
  color: #2B2C34;
}

.c7 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 2.156rem;
  text-transform: none;
  color: #2B2C34;
}

.c0 {
  width: 300px;
  height: 425px;
  padding: 1.4rem;
  background: #FFFFFE;
  position: relative;
  box-shadow: 4px 6px 13px rgba(0,0,0,0.1);
  border-radius: 12px;
  text-align: start;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  cursor: pointer;
}

.c0 > h1 {
  margin-left: 1rem;
}

.c0 img {
  margin-bottom: 2rem;
}

.c0:hover {
  -webkit-transform: scale(1.02);
  -ms-transform: scale(1.02);
  transform: scale(1.02);
  -webkit-filter: drop-shadow(0 1.5px 2px rgba(0,0,0,0.2));
  filter: drop-shadow(0 1.5px 2px rgba(0,0,0,0.2));
}

.c2 {
  position: absolute;
  padding: 0 1.4rem;
  width: 100%;
  height: 57px;
  left: 0px;
  top: 348px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #2B2C34;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 90%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  gap: 15px;
  color: #2B2C34;
}

.c4 {
  overflow: hidden;
  border-radius: 50%;
  width: 50px;
  height: 50px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 2px;
  gap: 4.2px;
  width: 184px;
  height: 46.2px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

.c6 > svg {
  color: #2B2C34;
  font-size: 2.4rem;
}

.c6 > h1 {
  display: inline;
}

@media (max-width:400px) {
  .c3 > img {
    width: 47px;
    height: 47px;
  }
}

<article
  class="c0"
  id="95380751"
>
  <img
    alt="aa"
    data-nimg="1"
    decoding="async"
    height="198"
    loading="lazy"
    src="https://www.datocms-assets.com/90479/1674243496-vuejs.svg"
    srcset="https://www.datocms-assets.com/90479/1674243496-vuejs.svg 1x, https://www.datocms-assets.com/90479/1674243496-vuejs.svg 2x"
    style="color: transparent;"
    width="260"
  />
  <h1
    class="c1"
    color="black"
  >
    Vue JS: O que Ã©, como funciona ?
  </h1>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <img
          alt="author image"
          data-nimg="1"
          decoding="async"
          height="50"
          src="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.datocms-assets.com%2F90479%2F1673505399-67812250_126640048653747_8066872119595701241_n.jpg&w=128&q=75 2x"
          style="color: transparent;"
          width="50"
        />
      </div>
      <div
        class="c5"
      >
        <h4>
          Mateus Duarte
        </h4>
        <p>
          02/03/2023
        </p>
      </div>
    </div>
    <div
      class="c6"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
        />
      </svg>
      <h1
        class="c7"
      >
        0
      </h1>
    </div>
  </div>
</article>
`;
